src/main/java/org/cloudfoundry/promregator/config/PromregatorConfiguration.kt

```diff
@@ -0,0 +1,88 @@
+package org.cloudfoundry.promregator.config
+
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.boot.context.properties.ConstructorBinding
+import java.util.*
+
+@ConstructorBinding
+@ConfigurationProperties(prefix = "promregator", ignoreInvalidFields = true)
+data class PromregatorConfiguration(
+
+        val workaround: WorkAroundConfig = WorkAroundConfig(),
+        val simulation: SimulationConfig = SimulationConfig(),
+        val reactor: ReactorConfig = ReactorConfig(),
+
+        var targets: List<Target> = ArrayList(),
+        var authenticator: AuthenticatorConfiguration = AuthenticatorConfiguration(),
```

Comment:
We should reconsider having this: This was only necessary to ensure compatibility. 

Background: Initially, there was only one authenticator planned. But it turned out shortly thereafter that we need more of them. Unfortunately, then the "single approach" was already out in the wild.

Do it here or shall we better create a new card at https://github.com/promregator/promregator/projects/2?

Reply:
I'll eliminate it here.